reference/format/datagrid.rst

```diff
@@ -0,0 +1,166 @@
+Datagrid
+========
++-----------+------------------+
+| Filename  | ``datagrid.yml`` |
++-----------+------------------+
+| Root Node | ``datagrid``     |
++-----------+------------------+
+
+The datagrid configuration is a large file that is a map of datagrid names mapped to options that
+configure their behavior:
+
+.. _reference-datagrid-columns:
+
+``actions`` (**type**: ``map``)
+
+    This a map of actions that can be performed from the datagrid. The keys are used to give unique
+    identifiers inside the grid, but do not have a special meaning. For each action, a map must be
+    passed configuring the action using the following options:
+
+    ``icon`` (**type**: ``string``)
+
+        The name of the Font Awesome icon that represents the action.
+
+    ``label`` (**type**: ``string``)
+
+        The action label that is displayed when the user hovers the icon. This will automatically
+        be translated.
+
+    ``link`` (**type**: ``string``)
+
+        The name of a link to associate with this action which is configured in the ``properties``
+        section.
+
+    ``rowAction`` (**type**: ``boolean`` **default**: ``false``)
+
+        When set to ``true``, the user can click anywhere on the row to perform this action.
+
+    ``type`` (**type**: ``string``)
+
+        This can either be ``navigate`` or ``delete``. When the value is ``delete``, an HTTP DELETE
+        request will be performed to the configured URL, otherwise the resource will be requested
+        with a GET request in the background and the response will be displayed.
+
+``columns`` (**type**: ``map``)
+
+    Each key of the map corresponds to a property of a select entry that should be displayed in the
+    grid. For each column you have to pass a map of options to configure how the column is shown:
+
+    ``choices`` (**type**: ``map``)
+
+        When using the ``select`` type, this options provides a mapping of stored values to their
+        human-readable representation.
+
+        When the ``type`` option is set to ``translatable``, this is treated as an expression that
+        will be evaluated to gain the value's translated representation.
+
+    ``frontend_type`` (**type**: ``string``)
+
+        The widget type that renders the value. Available types that are shipped with OroPlatform
+        by default are ``currency``, ``date``, ``datetime``, ``html``, ``integer``, ``select`` and ``string``.
```

Comment:
There are some types not defined here, see https://github.com/orocrm/platform/blob/16be2f85f66eb099cb42e211af680a79aaf8da81/src/Oro/Bundle/DataGridBundle/Resources/doc/backend/extensions/formatter.md#field